myst_nb/glue.py

```diff
@@ -0,0 +1,173 @@
+import nbformat as nbf
+from pathlib import Path
+import IPython
+from IPython.display import display as ipy_display
+from sphinx.util.docutils import SphinxDirective
+from sphinx.transforms import SphinxTransform
+from docutils import nodes
+from docutils.parsers.rst import directives
+from sphinx.util import logging
+
+from .parser import CellNode, CellInputNode, CellOutputBundleNode
+
+SPHINX_LOGGER = logging.getLogger(__name__)
+
+mime_prefix = "application/papermill.record/"
+
+
+def glue(name, variable):
```

Comment:
For things like figures, we probably want to show them in the notebook, at the same time as 'glueing' them, e.g. `glue("key", fig, display=True)`.

a) This would probably be used at the same time as tagging the cell with `remove_output`
b) In this use case we probably wouldn't want to prefix the mime types, because then we would have to store duplicate prefixed/non-prefixed outputs, so we would want to indicate whether the mime types had been prefixed, e.g.

`metadata["scrapbook"] = dict(name=name, mime_prefix=None)`

Reply:
yeah - my thinking was that people probably have already shown the figure, since it gets shown by default when you create the matplotlib object. At least when I've used scrapbook, I *always* end up having two figures plotted side-by-side, which is why I didn't feel the need to implement this in the current pass (since the act of "gluing" a plot is orthogonal to the act of "displaying" a plot). Do you feel strongly that this needs to be possible?

Reply:
Yes matplotlib is a bit of an outlier in this respect, in that basically it doesn't use the `IPython` mechanics properly. All other packages, like pandas, (correctly) don't have this 'auto-display' behaviour, and so this is where you would use it.
(you can also turn off matplotlib's auto show I think: https://stackoverflow.com/questions/18717877/prevent-plot-from-showing-in-jupyter-notebook)